src/constructors/constructWithOptions.js

```diff
@@ -11,6 +11,8 @@ export default (css: Function) => {
     }
 
     /* This is callable directly as a template function */
+    // TODO - try modifying this to automatically call extends
+    // and remove all extra wrapper components
```

Comment:
Hm, I think it's better to address this in a separate PR. It seems like we'll end up with lots of different approaches that'll have to be benchmarked against each other.

Not sure what you're addressing here regarding the wrapper calls? The extend method already merges all rules for the subsequent component